# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1219244484 14400
#      Wed Aug 20 11:01:24 2008 -0400
# Node ID 30d15ab0ce01de0998cea02261534167bd07cdce
# Parent  827d4f24ec6cdb6c22971326685ffd3d6d9eca0d
avoid problematic subfunction call in an anonymous function

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-08-20  Jaroslav Hajek <highegg@gmail.com>
+
+	* pkg/pkg.m (configure_make): Pass handle to is_architecture_dependent
+	directly.
+
 2008-08-20  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Don't set pm3d implicit if the plot
 	contains a surface. Fixes things like meshc(peaks()).
 
 2008-08-19  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/glpk.m: Fix invalid call to zeros.
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -1297,17 +1297,17 @@ function configure_make (desc, packdir, 
       endif
       filenames = split_by (filenames, " ");
     endif
 
     ## Split into architecture dependent and independent files
     if (isempty (filenames))
       idx = [];
     else
-      idx = cellfun (@(x) is_architecture_dependent (x), filenames);
+      idx = cellfun (@is_architecture_dependent, filenames);
     endif
     archdependent = filenames (idx);
     archindependent = filenames (!idx);
 
     ## Copy the files
     if (! all (isspace ([filenames{:}])))
 	if (! exist (instdir, "dir")) 
 	  mkdir (instdir);
